assets/js/alerts.js

```diff
@@ -0,0 +1,228 @@
+        : 'N/A'
+      const generatorLink = alert.generatorURL
+        ? `<a href="${alert.generatorURL}" class="statusTileHealth" target="_blank">View</a>`
+        : 'N/A'
+      $('#statusRows')
+        .append(`<tr data-alert-name="${alert.labels.application}" data-environment="${alert.labels.application}" data-environment-type="${alert.labels.environment}" id="tile-${alert.labels.application}-${alert.labels.environment}">
+          <td>${alert.labels.alertname}</td>
+          <td>${alert.annotations.message} </td>
+          <td>${dashboardLink}</td>
+          <td>${runbookLink} </td>
+          <td>${generatorLink}</td>
+        </tr>`)
+    })
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+function getOptions(data, key) {
+  const set = new Set(data.map(a => a.labels[`${key}`]))
+  return Array.from(set).sort()
+}
+
+function removeOptions(selectElement) {
+  var i,
+    L = selectElement.options.length - 1
+  for (i = L; i >= 0; i--) {
+    selectElement.remove(i)
+  }
+  const opt = document.createElement('option')
+  opt.value = ''
+  opt.textContent = ''
+  selectElement.appendChild(opt)
+}
+
+function renderDropdown(select, options, selectedValue, key) {
+  //clear dropDown options to prevent duplicates
+  removeOptions(select)
+  if (selectedValue != '') {
+    //append selected option
+    options.forEach(option => {
+      const opt = document.createElement('option')
+      opt.value = option
+      opt.textContent = option
+      if (option === selectedValue) opt.selected = true
+      select.appendChild(opt)
+    })
+  } else {
+    options.forEach(option => {
+      const opt = document.createElement('option')
+      opt.value = option
+      opt.textContent = option
+      select.appendChild(opt)
+    })
+  }
+}
+
+function populateAlertFilters(dropDownFilters) {
+  dropDownFilters.forEach(filter => {
+    filter.forEach(item => {
+      let select = document.getElementById(`${item.type}`)
+      select.add(new Option(item.text, item.text, false, item.selected))
+    })
+  })
+}
+
+function updateDropdowns(filteredData, currentFilters) {
+  // Get dynamic options from filtered data
+  const applications = getOptions(filteredData, 'application')
+  const environments = getOptions(filteredData, 'environment')
+  const namespaces = getOptions(filteredData, 'namespace')
+  const severities = getOptions(filteredData, 'severity')
+
+  renderDropdown(applicationFilter, applications, currentFilters.application, 'application')
+  renderDropdown(environmentFilter, environments, currentFilters.environment, 'environment')
+  renderDropdown(namespaceFilter, namespaces, currentFilters.namespace, 'namespace')
+  renderDropdown(severityFilter, severities, currentFilters.severity, 'severity')
+}
+
+//filter alters by selected filters
+function applyFilters(alerts, filters) {
+  return alerts.filter(alert => {
+    if (filters.application && alert.labels.application !== filters.application) return false
```

Comment:
Could just turn this into a big `return` statement and `||` the conditions of the multiple `if` statements.

Reply:
The same was as in `isFiltersEmpty`